### ISSUE TITLE
config is lost when start up is delayed

### ISSUE THREAD

**Issue by danielepolencic** [UKHomeOffice/hof-bootstrap#4]
When `startOnInitialise` is `false` when the interface is `start`ed, the same config has to be passed to the method otherwise it will fail:

```javascript
const path = require('path');
const bootstrap = require('hof-bootstrap');

bootstrap({
  startOnInitialise: false,
  routes: [{
    fields: path.resolve(__dirname, 'fields'),
    views: path.resolve(__dirname, 'views'),
    steps: {
      '/one': {
        next: '/two',
        fields: ['fname', 'lname']
      },
      '/two': {
        template: 'two'
      }
    }
  }]
}).then(bootstrap => {
  bootstrap.start(); // silently fails
  bootstrap.start({startOnInitialise: true}); // silently fails
  bootstrap.start({startOnInitialise: true, protocol: 'http', port: '8080', host: '0.0.0.0'}); // finally works
});
```

In the first 2 scenarios the app never starts (the `listen` method cannot be reached in the first, `port`, `host` and `protocol` are `undefined` in the last).

Perhaps we could a find a better idea to expose the interface. Perhaps we could expose the app as an express `Router`?

**Comment by JoeChapman** [comment]
we could remove the option to not start on initial bootstrap call, we could get extend the defaults with the config passed in to start every time? But I am curious to see what you have in mind re exposing the intercase

**Comment by danielepolencic** [comment]
I wanted to apply a middleware and the only way to do so was to use `use`. Also, I wanted my middleware to be applied before starting the app.

**Comment by JoeChapman** [comment]
ok, so removing the option `start` is off the table

**Comment by danielepolencic** [comment]
I think returning a non started interface could be an option.

**Comment by JoeChapman** [comment]
I think a non-started interface can be useful , esp. because of the scenario you have described re middleware. That is the reason I designed it as such, but I wonder how much people will need that feature. However, I have pushed a commit to the pr that is not particularly elegant, but addresses the issue by conditionally updating the config from the defaults and any new options the user might pass to `start`.
